internal/model/plugin.go

```diff
@@ -7,9 +7,12 @@ import (
 	mattermostModel "github.com/mattermost/mattermost-server/model"
 )
 
-// PluginSignatures represents Mattermost plugin signatures
-type PluginSignatures struct {
-	Signature     string `json:"signature"`
+// PluginSignature is a public key signature of a plugin and the corresponding public key hash for use in verifying a plugin downloaded from the marketplace.
+type PluginSignature struct {
+	// Signature represents a signature of a plugin saved in base64 encoding.
+	Signature string `json:"signature"`
+	// PublicKeyHash represents first arbitrary number of symbols of the
```

Comment:
Having trouble parsing this comment: what does `first arbitrary number of symbols` mean?

Reply:
We will check the suffix of the real public key hash. So arbitrary number of symbols might be in `PublicKeyHash` and it will still work. Our own convention will be to use 6 symbols.

Reply:
Would it make sense to document that `6` here instead? I'm wondering when that constant when change.

Reply:
In the rarest occasion when first 6 symbols of the two public key hashes are the same, we can use first 7 symbols for those keys. It's far-fetched, but still.

On the other hand maybe some of our customers who will have their own marketplace can adopt different number of symbols. They even can adopt variable number of symbols for different keys. They can use 1 symbol of a hash if there is a collision with previous keys add another symbol and so on.

Reply:
Maybe we can use language like https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection, i.e. `abbreviated public key hashes`.